models/src/main/protobuf/CasperMessage.proto

```diff
@@ -17,6 +17,7 @@ service DeployService {
   rpc createBlock(google.protobuf.Empty) returns (DeployServiceResponse) {}
   rpc showBlock(BlockQuery) returns (BlockQueryResponse) {}
   rpc showBlocks(google.protobuf.Empty) returns (BlocksResponse) {}
+  rpc listenForName(ListeningNameQuery) returns (ListeningNameResponse) {}
```

Comment:
I think it would be better to take a `Channel` directly, saves you one unnecessary level of `Option`. Also, is `Channel` the right type? That also includes `Var`s, which I don't think we want to be allowed to query for. What if this took a `Par` instead and constructed the `Channel` instance from it?